nuxeo-services/nuxeo-platform-content-template-manager/src/main/java/org/nuxeo/ecm/platform/content/template/factories/SimpleTemplateBasedFactory.java

```diff
@@ -76,22 +77,24 @@ protected void setProperties(List<PropertyDescriptor> properties, DocumentModel
     }
 
     protected void setAcl(List<ACEDescriptor> aces, DocumentRef ref) {
-        if (aces != null && !aces.isEmpty()) {
-            ACP acp = session.getACP(ref);
-            ACL existingACL = acp.getOrCreateACL();
-
-            // clean any existing ACL (should a merge strategy be adopted
-            // instead?)
-            existingACL.clear();
-
-            // add the the ACL defined in the descriptor
-            for (ACEDescriptor ace : aces) {
-                existingACL.add(new ACE(ace.getPrincipal(), ace.getPermission(), ace.getGranted()));
+        CoreInstance.doPrivileged(session, session -> {
```

Comment:
It's aways a good practice to let a comment explaining why we use a privilege session